src/pages/YearPickerPage.js

```diff
@@ -0,0 +1,110 @@
+        const minYear = Number(params.min);
+        const maxYear = Number(params.max);
+        const currentYear = Number(params.year);
+
+        this.currentYear = currentYear;
+        this.yearList = _.map(Array.from({length: (maxYear - minYear) + 1}, (v, i) => i + minYear), value => ({
+            text: value.toString(),
+            value,
+            keyForList: value.toString(),
+
+            // Include the green checkmark icon to indicate the currently selected value
+            customIcon: value === currentYear ? greenCheckmark : undefined,
+
+            // This property will make the currently selected value have bold text
+            boldStyle: value === currentYear,
+        }));
+
+        this.updateYearOfBirth = this.updateSelectedYear.bind(this);
+        this.filterYearList = this.filterYearList.bind(this);
+
+        this.state = {
+            inputText: '',
+            yearOptions: this.yearList,
+        };
+    }
+
+    /**
+     * Function called on selection of the year, to take user back to the previous screen
+     *
+     * @param {String} selectedYear
+     */
+    updateSelectedYear(selectedYear) {
+        // We have to navigate using concatenation here as it is not possible to pass a function as a route param
+        Navigation.navigate(`${this.props.route.params.backTo}?year=${selectedYear}`);
+    }
+
+    /**
+     * Function filtering the list of the items when using search input
+     *
+     * @param {String} text
+     */
+    filterYearList(text) {
+        this.setState({
+            inputText: text,
+            yearOptions: _.filter(this.yearList, year => year.text.includes(text.trim())),
+        });
+    }
+
+    render() {
+        return (
+            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+                <HeaderWithCloseButton
+                    title={this.props.translate('yearPickerPage.year')}
+                    shouldShowBackButton
+                    onBackButtonPress={() => Navigation.navigate(`${this.props.route.params.backTo}?year=${this.currentYear}` || ROUTES.HOME)}
+                    onCloseButtonPress={() => Navigation.dismissModal(true)}
+                />
+                <OptionsSelector
+                    textInputLabel={this.props.translate('yearPickerPage.selectYear')}
+                    onChangeText={this.filterYearList}
+                    keyboardType={CONST.KEYBOARD_TYPE.NUMBER_PAD}
+                    maxLength={4}
+                    value={this.state.inputText}
+                    sections={[{data: this.state.yearOptions, indexOffset: 0}]}
+                    onSelectRow={option => this.updateSelectedYear(option.value)}
+                    initiallyFocusedOptionKey={this.currentYear.toString()}
+                    hideSectionHeaders
+                    optionHoveredStyle={styles.hoveredComponentBG}
+                    shouldHaveOptionSeparator
+                    contentContainerStyles={[styles.ph5]}
+                />
```

Comment:
Here we added a Search option but failed to handle no results case for the search. 

Which lead to this [issue](https://github.com/Expensify/App/issues/16713#issue-1646503692)